packages/embed/src/utils/setup-google-analytics-instance.ts

```diff
@@ -38,9 +37,24 @@ const logError = (message: string) => {
   console.error(message)
 }
 
+const getTrackingFromDataLayer = (): string | undefined => {
+  if (window['dataLayer']) {
+    const config = window['dataLayer'].find((entry: string[]) => {
+      return entry.length > 1 && entry[0] === 'config'
+    })
+    return config && config[1]
+  }
+}
+
 export const setupGaInstance = (iframe: HTMLIFrameElement, embedId: string, shareGaInstance?: string | boolean) => {
-  const trackingId = typeof shareGaInstance === 'string' ? shareGaInstance : undefined
+  let trackingId = typeof shareGaInstance === 'string' ? shareGaInstance : undefined
   if (window.gtag) {
+    if (!trackingId) {
+      trackingId = getTrackingFromDataLayer()
+    }
+    if (!trackingId) {
+      return
```

Comment:
Maybe we could `logError` here in case this fails? It might indicate incorrect GTM setup in the host page and a log would help customers find out.